LogosLinuxInstaller.py

```diff
@@ -290,24 +299,33 @@ def main():
     utils.die_if_root()
 
     # Print terminal banner
-    msg.cli_msg(f"{config.LOGOS_SCRIPT_TITLE}, {config.LOGOS_SCRIPT_VERSION} by {config.LOGOS_SCRIPT_AUTHOR}.")
+    logging.info(f"{config.LOGOS_SCRIPT_TITLE}, {config.LOGOS_SCRIPT_VERSION} by {config.LOGOS_SCRIPT_AUTHOR}.")
+    logging.debug(f"Installer log file: {config.LOGOS_LOG}")
+
+    utils.self_update()
 
     # Check if app is installed.
-    if config.ACTION.__name__ == 'remove_install_dir' or config.ACTION.__name__ == 'utils.check_dependencies': # These actions don't require an install
+    if config.ACTION == "disabled":
+        msg.logos_error("That option is disabled.", "info")
+    elif config.ACTION.__name__ == 'remove_install_dir' or config.ACTION.__name__ == 'utils.check_dependencies':  # These actions don't require an install
         logging.info(f"Running function: {config.ACTION.__name__}")
         config.ACTION()
-    elif config.ACTION.__name__ == 'utils.set_appimage_symlink':
+    elif config.ACTION.__name__ == 'update_to_latest_recommended_appimage':
+        logging.info(f"Running function: {config.ACTION.__name__}")
+        utils.update_to_latest_recommended_appimage()
+    elif config.ACTION.__name__ == 'set_appimage_symlink':
         logging.info(f"Running function: {config.ACTION.__name__}")
         utils.set_appimage_symlink()
     elif utils.app_is_installed():
         # Run the desired Logos action.
         if config.ACTION is None:
             msg.logos_error("No function given for this subcommand.")
         logging.info(f"Running function: {config.ACTION.__name__}")
-        config.ACTION() # defaults to run_control_panel()
+        config.ACTION()  # defaults to run_control_panel()
     else:
         logging.info(f"Starting Control Panel")
         run_control_panel()
```

Comment:
The reason for the `if config.ACTION.__name__ == '...'` check is to let certain actions happen even if the app isn't fully installed, such as `remove_install_dir` and `check_dependencies`. So if there are additional actions that fall in that category, then I think we could streamline this by having a list of "pre-install" or "install-agnostic" actions. So we could end up with something like:
```python
install_not_required = [ 'action1', 'action2', 'actionN' ]
if config.ACTION.__name__ in install_not_required:
    logging.info(f"Running function: {config.ACTION.__name__}")
    config.ACTION()
elif utils.app_is_installed():
    # unchanged
else:
    # unchanged
```

Reply:
So the main addition to this function is the "disabled" line, in order to account for disabled functions. I'm guessing you are referring to lines 310, 313, and 316.

Those should be easy to fix. Am I understanding this comment properly?